### PR TITLE
[quant][graphmode][fx][bc-breaking] Support for reference pattern for fixqparam ops in eval mode

### PR DESCRIPTION
**Pull request by jerryzh168**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #62861
* #62698
* __->__ #62608

Summary:
Insert extra fixeqparam fake quant in the output of fixed qparam ops in fbgemm e.g. sigmoid
so that we can produce reference patterns for these ops

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps
Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D30053978](https://our.internmc.facebook.com/intern/diff/D30053978)

### CODE REVIEW

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30053978).

**Review by vkuzo — APPROVED** [review]

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30053978).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30053978).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30053978).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30053978).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30053978).

File: test/quantization/fx/test_numeric_suite_fx.py [hunk 1]
```diff
@@ -460,6 +460,7 @@ def forward(self, x0):
         self.assert_types_for_matched_subgraph_pairs(results, expected_types, mp, mq)
 
     @skipIfNoFBGEMM
+    @unittest.skip("Broken by https://github.com/pytorch/pytorch/pull/62608, need dtype inference support")
```
Comment (raghuramank100) [inline test/quantization/fx/test_numeric_suite_fx.py:463]: What is the ETA for dtype inference support?

**Review by raghuramank100 — APPROVED** [review]

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30053978).

File: test/quantization/fx/test_numeric_suite_fx.py (hunk 1 above)
Comment (jerryzh168) [reply test/quantization/fx/test_numeric_suite_fx.py:463]: @vkuzo will be working on that, I think maybe after he has some time, not sure about the ETAs..

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30053978).

**Comment by jerryzh168** [comment]
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30053978).

**Pull request closed without merging**